features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.drawable.Drawable
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+@Suppress("MagicNumber")
+internal object Base64LRUCache : Cache, ComponentCallbacks2 {
+    @Suppress("UnsafeThirdPartyFunctionCall") // maxMemory does not throw
+    private val CACHE_MEMORY_SIZE = (Runtime.getRuntime().maxMemory() / 8).toInt()
+    private val ON_LOW_MEMORY_SIZE = CACHE_MEMORY_SIZE / 2 // 50% size
+    private val ON_MODERATE_MEMORY_SIZE = (CACHE_MEMORY_SIZE / 4) * 3 // 75% size
+
+    private var cache: LruCache<Int, String> = object : LruCache<Int, String>(CACHE_MEMORY_SIZE) {
+        override fun sizeOf(key: Int?, value: String): Int {
+            return value.toByteArray().size
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {}
+
+    override fun onLowMemory() {
+        cache.evictAll()
+    }
+
+    @VisibleForTesting
+    internal fun setBackingCache(cache: LruCache<Int, String>) {
+        this.cache = cache
+    }
+
+    @Synchronized
+    override fun put(drawable: Drawable, base64String: String) {
+        val key = generateKey(drawable)
+        cache.put(key, base64String)
+    }
+
+    @Synchronized
+    override fun get(key: Int): String? {
+        return cache.get(key)
+    }
+
+    @Synchronized
+    override fun size(): Int = cache.size()
+
+    @Synchronized
+    override fun clear() {
+        cache.evictAll()
+    }
+
+    @Synchronized
+    override fun isEmpty(): Boolean =
+        cache.size() == 0
+
+    private fun generateKey(drawable: Drawable): Int =
+        System.identityHashCode(drawable)
```

Comment:
I'm wondering how we handle mutable drawables, like `ColorStateDrawable`. Obviously look of such drawable may be different (depending on the state), but hash code will be the same.

Or all the drawables coming to this layer are immutable?

Reply:
yes completely agree with @0xnm here. I think you need a reliable identifier / Drawable type.

Reply:
I'll go over the types of drawables and see if I can find a strategy of prepending data to the key. There are going to be some complicated cases like `AnimationDrawable` and `LayerDrawable`

Reply:
I've added prefixing to the keys for LayerDrawables and DrawableContainers